specification/configuration.md

```diff
@@ -89,6 +90,64 @@ environment variables:
   If `SPLUNK_CONFIG` is defined then these options are not required but MAY be
   defined.
 
+### Kubernetes Package Management Solutions
+
+While Kubernetes supports container technology that can be configured using
+environment variables, package management solutions such as Helm charts and
+Operators require YAML-based configuration. As a result, Kubernetes package
+management solutions MUST support the YAML configuration options specified
+below.
+
+> Any option description listed as REQUIRED means a value for the option MUST
+> be specified.
+
+- `clusterName` ()             : [REQUIRED] Name of the cluster.
+- `cloudProvider` ()           : Where Kubernetes is deployed.
+- `distribution` ()            : Which distribution of Kubernetes is deployed.
+- `environment` ()             : Name of the environment; if not defined then skipped.
+- `agent`                      : Deployed as a DaemonSet.
+  - `enabled` (`true`)         : Whether agent is deployed.
+  - `config` ()                : Updates configuration. Non-list options merged, list options override.
+- `gateway`                    : Deployed as a clustered Service and receives data from agent.
+  - `enabled` (`false`)        : Whether gateway is deployed.
+  - `config` ()                : Updates configuration. Non-list options merged, list options override.
+- `clusterReceiver`            : Deployed as a single replica deployment and collects Kubernetes API cluster and event telemetry.
+  - `enabled` (`true`)         : Whether k8sClusteReceiver is deployed. Ignored if `metricsEnabled` is `false`.
+  - `config` ()                : Updates configuration. Non-list options merged, list options override.
+
+In addition, at least one of the below configuration groups,
+`splunkObservability` or `splunkPlatform`, MUST be specified.
+
+> Any option description listed as REQUIRED means a value for the option MUST
+> be specified in that section. Both `accessToken` and `token` values are
+> stored as Kubernetes secrets. The secret key names are specified after this
+> section. If Kubernetes secret keys are specified they will override the
+> values specified below. In short, `accessToken` and `token` are required only
+> if not specified in Kubernetes secrets.
+
+- `splunkObservability`
```

Comment:
nit: generally config like this should be namespaced/encapsulated so using `splunk` prefix is probably redundant. Can we call these just `observability` and `platform/enterprise`?

Reply:
It's been a while since I used helm, so I might be missing something, but as I remember it the namespace is defined by the operator when they install the chart. Meaning it is up to them to correctly define it as `splunk`, right? If so, it might make sense to keep the prefix, even though it is redundant, to ensure a clear scope to the variable.

Reply:
It's not really a prefix for Splunk specific configuration, but rather part of a destination configuration. Full names for the destinations/products are "Splunk Observability" and "Splunk Platform" (https://www.splunk.com/en_us/software.html). `observability` may be self-descriptive, but just `platform` is confusing I think. I would keep the full names to make it clear.

Reply:
That sounds reasonable and not a blocker for me but to answer the question,

> It's been a while since I used helm, so I might be missing something, but as I remember it the namespace is defined by the operator when they install the chart. Meaning it is up to them to correctly define it as splunk, right? 

My bad for using the term "namespace". I did not mean a k8s namespace but namespace in a general sense.

I was thinking of somethig like this:

```
k get splunk-agent

kind: SplunkOtelAgent
spec:
    clusterName: ...
    splunkObservability: ...
    splunkPlatform: ...
```

here repeating the word splunk in config feels unnecessary. 

Is this not the case for helm? Does splunk config get merged into an object that contains non-splunk config?

Reply:
The configuration of the helm is applied to the whole chart release, it's on the root level, not a part of any object. There are some Splunk specific configs and Kubernetes specific configs in it. We already discussed extra grouping in this thread https://github.com/signalfx/gdi-specification/pull/147#discussion_r742334481. And I think we agreed that extra grouping like `splunkCommon` will make the user experience worse.

Looking at the same configuration in the `SplunkOtelAgent` k8s object, I think it's Ok to have some extra Splunk words in it given that `splunkPlatform` and `splunkObservability` refer to exact product names that IMO shouldn't be broken down.